azerothcore/azerothcore-wotlk
# Fix(Core/BWL): Vaelastrasz Burning Adrenaline

## Changes Proposed:
-  Merge the events into 1 and keep counting. Fixed a bug with target selector.
-  Fixed instant cast time when allowed, this somehow got removed by X commit (it magically appeared after a file rename...)
-  Added the spell instakill effect when the debuff is removed.

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/10582
- Closes https://github.com/chromiecraft/chromiecraft/issues/2945

## SOURCE:


## Tests Performed:

- Tested in-game.

## How to Test the Changes:


1. Go to BWL, Vaelastrasz with 2 or more chars (specially with mana).
2. Engage
3. Wait 15 seconds, you should see your not tank char has the debuff.

## Known Issues and TODO List:


- [ ] Didn't found anything wrong with the damage.


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
